autorest/codegen/templates/lro_operation.py.jinja2

```diff
@@ -91,7 +91,11 @@ response_headers = {
         'polling_interval',
         self._config.polling_interval
     )
+    {% if code_model.options['azure_arm'] %}
     if polling is True: polling_method = {{ async_prefix }}ARMPolling(lro_delay{{ lro_options }},  **kwargs)
+    {% else %}
+    if polling is True: raise ValueError("You are attempting to use ARM polling for a non-ARM operation. Set polling to False for non ARM polling")
```

Comment:
Actually, from customers point of view, they don't know what ARM is. I would raise something more:
> There is no default defined polling implemetation, and then True is not a valid value

Since actually True means "use the default one"

We might add later something like `--default-polling=azure.cognitiveservices.polling.CSPolling`, where the default value in ARM mode is `azure.mgmt.core.polling.ArmPolling`